src/document/UBSortFilterProxyModel.cpp

```diff
@@ -23,3 +23,19 @@ bool UBSortFilterProxyModel::lessThan(const QModelIndex &left, const QModelIndex
 
     return QSortFilterProxyModel::lessThan(left, right);
 }
+
+bool UBSortFilterProxyModel::filterAcceptsRow(int sourceRow,
+                                              const QModelIndex &sourceParent) const
+{
+    QModelIndex index0 = sourceModel()->index(sourceRow, 0, sourceParent);
+    if(UBPersistenceManager::persistenceManager()->mDocumentTreeStructureModel->isCatalog(index0))
```

Comment:
Instead of using the `UBPersistenceManager` to get the model you should better use the same code as in `lessThan()`, cast the result of `sourceModel()` and check for `nullptr`. This avoids the implicit assumption that the model, where this filter works upon, is exactly that `mDocumentTreeStructureModel` managed by the `UBPersistenceManager`.

See also below for removing the include of the `UBPersistenceManager` in the header.

Reply:
Done